src/__tests__/use-document/use-document.test.tsx

```diff
@@ -91,4 +103,52 @@ describe('_document support', () => {
       });
     }
   );
+
+  describe('next/head', () => {
+    describe('first render', () => {
+      it('merges _document and page head elements', async () => {
+        const { page } = await getPage({
+          nextRoot: path.join(
+            __dirname,
+            '/__fixtures__/custom-document-with-gip'
+          ),
+          route: '/page',
+          useDocument: true,
+        });
+        const { container } = render(page);
+
+        const metaDescriptions = getMetaTagsContentByName(
+          container,
+          'description'
+        );
+        expect(metaDescriptions.length).toBe(2);
+        expect(metaDescriptions[0]).toBe('Custom document description');
+        expect(metaDescriptions[1]).toBe('Page description');
+      });
+    });
+
+    describe('on client navigation', () => {
+      it('merges _document and page head elements', async () => {
+        const { page } = await getPage({
+          nextRoot: path.join(
+            __dirname,
+            '/__fixtures__/custom-document-with-gip'
+          ),
+          route: '/page',
+          useDocument: true,
+        });
+        const { container } = render(page);
+        userEvent.click(screen.getByText('Go to A'));
+        await screen.findByText('This is page A');
+
+        const metaDescriptions = getMetaTagsContentByName(
+          container,
+          'description'
+        );
+        expect(metaDescriptions.length).toBe(2);
+        expect(metaDescriptions[0]).toBe('Custom document description');
+        expect(metaDescriptions[1]).toBe('Page A description');
```

Comment:
Will there be multiple "descriptions" in NextJS app as well? I think so; just want to make sure.

Reply:
Yep, I was surprised to see that `next/head` simply pushes its meta elements with the ones declared in `_document`.